pkg/server/router/route_appender_factory.go

```diff
@@ -23,15 +23,18 @@ func NewRouteAppenderFactory(staticConfiguration static.Configuration, entryPoin
 type RouteAppenderFactory struct {
 	staticConfiguration static.Configuration
 	entryPointName      string
-	acmeProvider        *acme.Provider
+	acmeProvider        []*acme.Provider
 }
 
 // NewAppender Creates a new RouteAppender
 func (r *RouteAppenderFactory) NewAppender(ctx context.Context, middlewaresBuilder *middleware.Builder, runtimeConfiguration *runtime.Configuration) types.RouteAppender {
 	aggregator := NewRouteAppenderAggregator(ctx, middlewaresBuilder, r.staticConfiguration, r.entryPointName, runtimeConfiguration)
 
-	if r.acmeProvider != nil && r.acmeProvider.HTTPChallenge != nil && r.acmeProvider.HTTPChallenge.EntryPoint == r.entryPointName {
-		aggregator.AddAppender(r.acmeProvider)
+	for _, p := range r.acmeProvider {
+		if p != nil && p.HTTPChallenge != nil && p.HTTPChallenge.EntryPoint == r.entryPointName {
+			aggregator.AddAppender(p)
+			break
```

Comment:
Shouldn't we log about the fact that there are several acmeProviders and we ignore all of them except for the first one?